src/orange/OrangeTrackView.hh

```diff
@@ -234,15 +251,26 @@ CELER_FUNCTION
 OrangeTrackView& OrangeTrackView::operator=(DetailedInitializer const& init)
 {
     CELER_EXPECT(is_soft_unit_vector(init.dir));
-    CELER_EXPECT(states_.vol[init.other.thread_]);
+
+    for (auto i : range(states_.level[init.other.thread_] + 1))
+    {
+        auto lsa = this->make_lsa(thread_, LevelId{i});
+        auto lsa_other = this->make_lsa(init.other.thread_, LevelId{i});
+
+        CELER_EXPECT(lsa_other.vol());
+
+        lsa.vol() = lsa_other.vol();
+        lsa.pos() = lsa_other.pos();
+        lsa.dir() = init.dir;
+        lsa.universe() = lsa_other.universe();
+        lsa.surf() = lsa_other.surf();
+        lsa.sense() = lsa_other.sense();
+        lsa.boundary() = lsa_other.boundary();
```

Comment:
Maybe it would make sense to define an `operator=` to LSA that copies the underlying data? It would reduce the (growing) amount of code in this class.

Reply:
In other words you would be able to do:

```
lsa = lsa_other
```

And all the fields would be copied under the hood?

Reply:
Exactly.